packages/kit/src/core/create_app/index.js

```diff
@@ -130,11 +130,13 @@ function generate_app(manifest_data) {
 
 	while (l--) {
 		pyramid = `
-			<svelte:component this={components[${l}]} {...(props_${l} || {})}>
-				{#if components[${l + 1}]}
+			{#if components[${l + 1}]}
+				<svelte:component this={components[${l}]} {...(props_${l} || {})}>				
 					${pyramid.replace(/\n/g, '\n\t\t\t\t\t')}
-				{/if}
-			</svelte:component>
+				</svelte:component>
+			{:else}
+				<svelte:component this={components[${l}]} {...(props_${l} || {})} />
+			{/if}
```

Comment:
I used to do this trick in my other projects, but a caveat I found is that this would also rerender the component if only the child components change, which is sometimes undesirable, especially for route transitions. And so, I think we should fix the warning in Svelte side, and leave the code as is for now to prevent a potential breaking change.

Reply:
When does that happen? Are there situations where the upper level component stays the same but the underlying level changes to falsy?

Reply:
I'm also curious about this — if we can fix it here rather than in Svelte, that would be preferable I think, since the warning is useful.

The `components` 'branch' is an array of nodes, where every node except the leaf is a layout. Layouts always have a slotted child, leaves never do. Which means that I don't think there's ever a scenario in which a component could move between the consequent and the alternate — layouts will _always_ be in the consequent, leaves will _always_ be in the alternate.

In other words this is a brilliantly simple (and in hindsight, obvious!) fix for this issue as far as I can see — thank you @bfanger! But I'd love to better understand @bluwy's concern before merging.

Reply:
Ah thanks for the explanation. I wasn't sure what the `components` array would contain, but if the quirk I mention never happens. This looks good to me!